### ISSUE TITLE
the weights of the yolo26 model

### ISSUE THREAD

**Issue by ctx289** [ultralytics/ultralytics#23188]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Where can I download the weights of the yolo26 model that does not use an end-to-end detection head? I found that directly using the end-to-end YAML configuration file for data recall training yields rather poor performance.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ctx289, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like finding the right YOLO26 weights/head configuration and reconciling training performance), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 📎🧪

Also, could you share a bit more context so we can route this correctly:
-Which YOLO26 YAML file(s) are you using (end-to-end vs non end-to-end) and where are they located in your setup?
-What `ultralytics` version are you on, and what command (CLI) or snippet (Python) are you using to train?
-Please include key metrics (mAP/precision/recall), your training logs, and a couple of representative images/labels if possible

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

> 🤖 This is an automated response to help gather the right details quickly—an Ultralytics engineer will also assist you soon.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by ctx289** [comment]
I suggest that the homepage could support direct downloads of these two types of weights to better meet users' needs.

**Comment by Y-T-G** [comment]
They use the same YAML and weights. The difference is during inference for end-to-end, you use one2one head. But for NMS based version, you use one2many head.

**Comment by ctx289** [comment]
<img width="997" height="259" alt="Image" src="https://github.com/user-attachments/assets/42ca80c5-69b1-4aee-8ed5-b4228f8fe937" />
By the way, I would like to ask what the current training configuration for YOLOv26 is? The corresponding parameters I printed out seem not very clear.

**Comment by Laughing-q** [comment]
@ctx289 to make the Detect head compatible with previous non-e2e YOLO models like YOLOv8/11 we had to make this `end2end` dynamically checking if there's `one2one` head.
https://github.com/ultralytics/ultralytics/blob/f2d3aed634a5b0e4828024718d4a61ab2f83fb19/ultralytics/nn/modules/head.py#L126-L128
The simplest way to run the one2many branch would be deleting the one2one head for now.
```python
model = YOLO("yolo26n.pt)
del model.model.model[-1].one2one_cv2
```

**Comment by Laughing-q** [comment]
> <img alt="Image" width="997" height="259" src="https://private-user-images.githubusercontent.com/100291372/535511550-42ca80c5-69b1-4aee-8ed5-b4228f8fe937.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgzODE2NTcsIm5iZiI6MTc2ODM4MTM1NywicGF0aCI6Ii8xMDAyOTEzNzIvNTM1NTExNTUwLTQyY2E4MGM1LTY5YjEtNGFlZS04ZWQ1LWI0MjI4ZjhmZTkzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDExNFQwOTAyMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NDRhN2I0NzhkZGYxYzJmNGQwODQ4ZTk0M2M1OTliYzE2YmE3NDQ1NzQ5NDEyNDUyYTBmY2E4MmQwN2U2MDZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lG6KgGCfcQUqE4gqGPwXDK0GnbhVSv5dpLvFlGBSYUI"> By the way, I would like to ask what the current training configuration for YOLOv26 is? The corresponding parameters I printed out seem not very clear.

it's exactly as it's showing in your screenshot, we leveraged our internal hyp search to further boost the model performance. What's showing is the hyps we are using for corresponding model training.

**Comment by ctx289** [comment]
> > <img alt="图像" width="997" height="259" src="https://private-user-images.githubusercontent.com/100291372/535511550-42ca80c5-69b1-4aee-8ed5-b4228f8fe937.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjgzODE2NTcsIm5iZiI6MTc2ODM4MTM1NywicGF0aCI6Ii8xMDAyOTEzNzIvNTM1NTExNTUwLTQyY2E4MGM1LTY5YjEtNGFlZS04ZWQ1LWI0MjI4ZjhmZTkzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDExNFQwOTAyMzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NDRhN2I0NzhkZGYxYzJmNGQwODQ4ZTk0M2M1OTliYzE2YmE3NDQ1NzQ5NDEyNDUyYTBmY2E4MmQwN2U2MDZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lG6KgGCfcQUqE4gqGPwXDK0GnbhVSv5dpLvFlGBSYUI">顺便问一下，YOLOv26 目前的训练配置是什么？我打印出来的相应参数似乎不太清楚。
> 
> 正如你截图中所示，我们利用内部的hyp搜索进一步提升了模型性能。截图中显示的是我们用于相应模型训练的hyp。

Thank you for your response. The current performance and work are absolutely outstanding! I believe this is a truly great revolution for YOLO, with better accuracy and higher speed. Could you please provide the training configuration that allows me to reproduce the current accuracy?

**Comment by yang-0201** [comment]
Is there a potential issue with the current YOLOv26n pretrained weights? When running validation using the command
yolo val detect data=coco.yaml device=0, the reported COCO mAP is 40.0, while the mAP evaluated using VOC format is 39.9.

**Comment by Laughing-q** [comment]
@yang-0201 by default it's doing end2end evaluation using the one2one head, which the result for nano is 40.1.

<img width="1098" height="619" alt="Image" src="https://github.com/user-attachments/assets/878c37e5-366e-4679-b7c7-72f5e82f3fe8" />

```python
from ultralytics import YOLO

weight = "yolo26n.pt"
model = YOLO(weight)
# del model.model.model[-1].one2one_cv2
results = model.val(data="coco.yaml", conf=0.0001, batch=128)
```
you can reproduce the results using the script above for end2end mAP, and for the non-e2e one currently we don't have such out-of-box option to do so, in the code internally we check the if go to the end2end path by checking if the mode got one2one modules, hence for now you need to manually delete the one2one head (by uncomment the line of `del model.model.model[-1].one2one_cv2`) to get the non-e2e results if you want.

<img width="1277" height="1292" alt="Image" src="https://github.com/user-attachments/assets/6efda64b-1ca9-49af-94ca-963ca5241c67" />

**Comment by Y-T-G** [comment]
You can run with `batch=1`

**Comment by yang-0201** [comment]
@Laughing-q  Thank you for your explanation. In addition, I noticed that after adopting the new optimizer, the YOLOv26n model was trained for only 245 epochs, and the small model reached such high accuracy with only 70 epochs, whereas YOLOv11 typically requires around 600 epochs of training.

Have you experimented with training the YOLOv26 models for a larger number of epochs? If so, does extending the training schedule further improve performance?

now I find this model is finetun-tune from yolo26-objv1-150，how to get them，I guess from object365？

**Comment by yang-0201** [comment]
In addition, I noticed that when training on the COCO dataset with the current codebase, the SGD optimizer is used by default. If muSGD is adopted instead, do you have any recommendations for setting the optimizer hyperparameters?

**Comment by glenn-jocher** [comment]
@yang-0201 glad it’s useful (credit to the Ultralytics team + broader YOLO community), but we don’t currently publish the intermediate pretrain checkpoints like `yolo26-objv1-150` (or the full internal hyp-search recipe) needed to exactly reproduce the released COCO numbers; the reproducible starting point is the public `yolo26*.pt` weights in the [Ultralytics YOLO26 docs](https://docs.ultralytics.com/models/yolo26/), and from there you can fine-tune and extend `epochs` for your dataset (COCO typically plateaus quickly, so watch `val` and stop when it stops improving).  

On MuSGD: the public trainer’s `optimizer` switch currently only supports the built-ins (see `BaseTrainer.build_optimizer()` in the [trainer reference](https://docs.ultralytics.com/reference/engine/trainer/)), so seeing SGD by default is expected; MuSGD is implemented as `ultralytics.optim.muon.MuSGD` (see the [MuSGD reference](https://docs.ultralytics.com/reference/optim/muon/)) but isn’t wired into `yolo train` yet—if you want to experiment, you can patch `build_optimizer()` to return MuSGD, e.g.:

```python
from ultralytics.optim.muon import MuSGD

optimizer = MuSGD(
    [
        {"params": g[2], "weight_decay": 0.0, "use_muon": True},  # biases
        {"params": g[0], "weight_decay": decay, "use_muon": True},  # weights
        {"params": g[1], "weight_decay": 0.0, "use_muon": True},  # norm weights
    ],
    lr=lr,
    momentum=momentum,
    nesterov=True,
    muon=0.5,
    sgd=0.5,
)
```

**Comment by yang-0201** [comment]
@glenn-jocher Thank you for your patient reply.

**Comment by Laughing-q** [comment]
> [@Laughing-q](https://github.com/Laughing-q) Thank you for your explanation. In addition, I noticed that after adopting the new optimizer, the YOLOv26n model was trained for only 245 epochs, and the small model reached such high accuracy with only 70 epochs, whereas YOLOv11 typically requires around 600 epochs of training.
> 
> Have you experimented with training the YOLOv26 models for a larger number of epochs? If so, does extending the training schedule further improve performance?
> 
> now I find this model is finetun-tune from yolo26-objv1-150，how to get them，I guess from object365？

yeah because the final training results are fine-tuned on pretrained weights from object365 so it dramatically reduced the training epochs we need. In our exps longer epochs ended up with overfitting when starting from pretrained weights.
You can see all the training hyps by checking the "train_args" we saved in the pt files.
One correction, we already open-sourced the object365 pretrained weights as well, you can check them all here: https://github.com/ultralytics/assets/releases/tag/v8.4.0

Apart from the topic, I also want to mention we used to train all our previous models from scratch for like 500 epochs for YOLO11 to get a decent results, which need several days to get one round experiment results. The needs for long training epochs reduced by starting from pretrained weights and accelerate our R&D process a lot(also boost accuracy) so we eventually stick to this approach.
Also worth to mention that with our MuSGD it accelerate training convergence a lot even training from scratch, for example we had a yolo26s training with MuSGD from scratch for 300 epochs beats training with default SGD for 500 epochs.

**Comment by Laughing-q** [comment]
> In addition, I noticed that when training on the COCO dataset with the current codebase, the SGD optimizer is used by default. If muSGD is adopted instead, do you have any recommendations for setting the optimizer hyperparameters?

I'm working on a [PR](https://github.com/ultralytics/ultralytics/pull/23234) to use MuSGD by default now when `optimizer=auto`, going to merge it after some tests in today.

**Comment by ctx289** [comment]
> > [@Laughing-q](https://github.com/Laughing-q) Thank you for your explanation. In addition, I noticed that after adopting the new optimizer, the YOLOv26n model was trained for only 245 epochs, and the small model reached such high accuracy with only 70 epochs, whereas YOLOv11 typically requires around 600 epochs of training.
> > Have you experimented with training the YOLOv26 models for a larger number of epochs? If so, does extending the training schedule further improve performance?
> > now I find this model is finetun-tune from yolo26-objv1-150，how to get them，I guess from object365？
> 
> yeah because the final training results are fine-tuned on pretrained weights from object365 so it dramatically reduced the training epochs we need. In our exps longer epochs ended up with overfitting when starting from pretrained weights. You can see all the training hyps by checking the "train_args" we saved in the pt files. One correction, we already open-sourced the object365 pretrained weights as well, you can check them all here: https://github.com/ultralytics/assets/releases/tag/v8.4.0
> 
> Apart from the topic, I also want to mention we used to train all our previous models from scratch for like 500 epochs for YOLO11 to get a decent results, which need several days to get one round experiment results. The needs for long training epochs reduced by starting from pretrained weights and accelerate our R&D process a lot(also boost accuracy) so we eventually stick to this approach. Also worth to mention that with our MuSGD it accelerate training convergence a lot even training from scratch, for example we had a yolo26s training with MuSGD from scratch for 300 epochs beats training with default SGD for 500 epochs.

Thank you for your patient and detailed response. That is to say, if I use the object detection pre-trained weights for my COCO80 dataset and follow the parameters in the aforementioned print settings, will I be able to reproduce these results?

**Comment by Laughing-q** [comment]
yes! oh noticed there might be some hyps/args in the train_args but we eventually removed for simplicity for now. the results might be slightly different the official numbers might should be the same level.
